app/DashBoard/Presenters/HomePagePresenter.php

```diff
@@ -18,26 +18,92 @@ class HomePagePresenter extends BasePresenter
 	) {
 		$this->projectControlFactory = $projectControlFactory;
 		$this->projectsRepository = $projectsRepository;
+		$this->usersFavoriteProjectsRepository = $usersFavoriteProjectsRepository;
+		$this->userRepository = $userRepository;
 	}
 
+	public function actionDefault()
+    {
+        $favoriteProjects = $this->usersFavoriteProjectsRepository->findBy(["user" => $this->getUser()->id])->orderBy("this->projects->name");
+
+        /** @var \Pd\Monitoring\UsersFavoriteProjects\UsersFavoriteProjects $favoriteProject */
+		foreach ($favoriteProjects as $favoriteProject) {
+            $this->favoriteArray[$favoriteProject->projects->id] = $favoriteProject;
+            $this->projects[$favoriteProject->projects->id] = $favoriteProject->projects;
+        }
+
+        $allProjects = $this->projectsRepository->findBy(["id!" => array_keys($this->favoriteArray)])->orderBy("name");
+        /** @var \Pd\Monitoring\Project\Project $project */
+        foreach ($allProjects as $project) {
+            $this->projects[$project->id] = $project;
+            $this->nonFavoriteProjects[$project->id] = $project;
+        }
+
+    }
 
 	public function renderDefault()
-	{
-		$this->template->projects = $this->projectsRepository->findAll()->orderBy('name');
-	}
+    {
+        $this->template->projects = $this->nonFavoriteProjects;
+        $this->template->favoriteProjects = $this->favoriteArray;
+    }
+
+    public function handleDeleteFavoriteProject(int $id)
+    {
+        $userId = $this->getUser()->id;
+        if($this->usersFavoriteProjectsRepository->checkIfUserHasFavoriteProject($userId, $id)) {
+            $this->usersFavoriteProjectsRepository->deleteFavoriteObject($userId, $id);
+            $this->flashMessage("Oblíbená položka byla odebrána.");
+        } else {
+            $this->flashMessage("Zadaná položka už byla smazána.");
+        }
+        $this->redirect("this");
+    }
 
+    public function handleSetFavoriteProject(int $id)
+    {
+        $userId = $this->getUser()->id;
+        if(!$this->usersFavoriteProjectsRepository->checkIfUserHasFavoriteProject($userId, $id)){
+            $favoriteProject = new \Pd\Monitoring\UsersFavoriteProjects\UsersFavoriteProjects();
+            $favoriteProject->user = $this->userRepository->findById($userId)->fetch();
+            $favoriteProject->projects = $this->projectsRepository->findById($id)->fetch();
+            $this->usersFavoriteProjectsRepository->persistAndFlush($favoriteProject);
+            $this->flashMessage("Oblíbená položka byla přidána.");
+        } else {
+            $this->flashMessage("Zadaná položka se už v oblíbených nachází.");
+        }
+        $this->redirect("this");
+    }
 
 	protected function createComponentProject() : \Nette\Application\UI\Multiplier
 	{
 		$cb = function ($id) {
-			$project = $this->projectsRepository->getById($id);
-			$control = $this->projectControlFactory->create($project);
+            $project = $this->projects[$id];
```

Comment:
Pokud by to nepřekročilo 120 znaků, volal bych to napřímo přímo v místě využití. Proměnná se už jinde nevyužije.